modules/Saritasa.Upsource/Saritasa.Upsource.psm1

```diff
@@ -0,0 +1,367 @@
+        [string]$Value
+    )
+
+    $textBytes = [System.Text.Encoding]::UTF8.GetBytes($value)
+
+    $encodedText = [Convert]::ToBase64String($textBytes)
+
+
+    return $encodedText
+}
+
+<#
+    Credentials encoding helper
+ #>
+function CredentialsEncoded {
+
+    $value = $UpsourceCredentials.UserName + ":" + $UpsourceCredentials.GetNetworkCredential().Password
+    return EncodeBase64 -Value $value
+}
+
+<#
+    Web request helper
+ #>
+function Invoke-Request {
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string]$Url,
+
+        [Parameter(Mandatory = $true)]
+        [string]$Body
+    )
+
+    $credentialsEncoded = CredentialsEncoded
+
+    Invoke-RestMethod  -ContentType 'application/json' -Headers @{'Authorization' = "Basic $credentialsEncoded"} `
+        -Body $Body `
+        -Method Post `
+        -Uri $Url `
+
+    return
+}
+
+<#
+    Get all revisions for project
+ #>
+function Get-RevisionsList {
+
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string]$Project,
+
+        [int]$Limit = [int]::MaxValue
+    )
+
+    $revisionListDto = @{
+        projectId    = $Project
+        limit        = $Limit
+        requestGraph = $false
+    } | ConvertTo-Json
+
+    $url = "$UpsourceUrl/~rpc/getRevisionsList"
+
+    $revisionList = Invoke-Request -Url $url -Body $revisionListDto
+
+    $revisionIds = @()
+
+    $revisionList.result.revision | ForEach-Object { 
+
+        if ($_ -ne $null) {
+
+            $revisionObject = @{  
+                RevisionId    = $_.revisionId 
+                Date          = $_.revisionDate
+                CommitMessage = $_.revisionCommitMessage
+                Author        = $_.authorId
+            }
+
+            $revisionIds += $revisionObject
+        }
+    }
+
+    $revisionIds
+}
+
+function Get-RevisionInfo {
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string]$ProjectId,
+
+        [Parameter(Mandatory = $true)]
+        [string]$RevisionId
+    )
+
+    $url = "$UpsourceUrl/~rpc/getRevisionInfo"
+
+    $params = @{
+        projectId  = $ProjectId
+        revisionId = $RevisionId
+    } | ConvertTo-Json
+
+    Invoke-Request -Url $url -Body $params
+}
+
+function Get-RevisionListFiltered {
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string]$ProjectId,
+
+        [int]$Limit = 30, 
+
+        [string]$Query
+    )
+
+    $requestDto = @{
+        projectId    = $ProjectId
+        limit        = $Limit
+        query        = $Query
+        requestGraph = $false
+    } | ConvertTo-Json
+
+    $url = "$UpsourceUrl/~rpc/getRevisionsListFiltered"
+
+    $revisionIds = @()
+
+    $result = Invoke-Request -Url $url -Body $requestDto
+
+    $result.result.revision | ForEach-Object { $revisionIds += $_.revisionId }
+
+    $revisionIds
+}
+
+function Get-RevisionsInReview {
```

Comment:
Let's use the same style of plural/singular names.

Get-RevisionList
Get-RevisionWithoutReview
Get-RevisionInReview